### ISSUE TITLE
provider-terraform-aws reconciler implementation

### ISSUE THREAD

**Issue by kasey** [crossplane/crossplane#1993]
### What problem are you facing?

We need to show how the schema types generated in https://github.com/crossplane/crossplane/issues/1844 will be used in a crossplane reconciler. This means having `provider-terraform-aws` implementations for all the [terraform-provider-runtime](https://github.com/crossplane-contrib/terraform-runtime/pull/1/files) interfaces used in the [provider-terraform-gcp](https://github.com/crossplane-contrib/provider-terraform-gcp/pull/1/files#diff-653ac63623bb610ba195fcb6396fe32ce42a8573019bc63d0af92ec9907c1197R16) prototype, potentially for a limited subset of aws resources.

### How could Crossplane help solve your problem?

There are 2 routes to get here.

One is finishing code generation for terraform<>crd serialization/deserialization so that reconcilers can be built for a large set of resources. Assuming that there would be edge cases in code generation, we would take the output of the code generation process and build an inclusion list of resources that are known to be operational. A benefit of this approach is that it would de-risk code generation by proving that our understanding of the translation process is correct and a complete implementation is feasible. But it also pushes integration to the end of the project and risks functioning code generation without a working reconciler.

We could also hand-write the serialization/deserialization code to demonstrate what the final result would look like. Unlike ACK, we don't have a code generation pipeline to build on, and we wouldn't normally write this kind of code, so this would be throw-away work. But it allows us to get earlier feedback on the overall UX and figure out integration with terraform-provider-runtime in order to get a working prototype sooner. Having already done this for terraform-provider-gcp, we think it should be relatively low effort to write resources this way. It does however delay completing the most difficult component of this project, which is lot easier to do with existing momentum.

I want to split the difference between these approaches, by just writing 1-3 resources by hand. In parallel I plan to wrap up the terraform->crd deserialization. This re-orders the final steps of this project by moving the integration pieces ahead of the last bit of R&D work so that we can show an end-to-end demo a little sooner but leaves time to make progress on codegen. A good candidate for the hand-written resource would be an IAM user, because this is a relatively simple resource, and ACK specifically plans to exclude it.

**Comment by negz** [comment]
We're experimenting with a different approach on this one at the moment in https://github.com/crossplane-contrib/terrajet - we're using that repo to track.
